packages/node/src/e2e/e2e-test.spec.ts

```diff
@@ -0,0 +1,125 @@
+// the environment wasn't properly set.
+describe("Environment", () => {
+  it("contains the expected environment variables", () => {
+    expect(process.env.REFRESH_TOKEN).not.toBeUndefined();
+    expect(process.env.CLIENT_ID).not.toBeUndefined();
+    expect(process.env.CLIENT_SECRET).not.toBeUndefined();
+  });
+});
+
+describe("Authenticated fetch", () => {
+  it("can fetch a public resource when logged in", async () => {
+    const session = new Session();
+    await session.login({
+      clientId: process.env.CLIENT_ID,
+      clientSecret: process.env.CLIENT_SECRET,
+      refreshToken: process.env.REFRESH_TOKEN,
+      oidcIssuer: OIDC_ISSUER,
+    });
+    const response = await session.fetch(
+      "https://ldp.demo-ess.inrupt.com/105177326598249077653/profile/card#me"
+    );
+    expect(response.status).toEqual(200);
+    await expect(response.text()).resolves.toContain(
+      "foaf:PersonalProfileDocument"
+    );
+  });
+
+  it("can fetch a private resource when logged in", async () => {
+    const session = new Session();
+    await session.login({
+      clientId: process.env.CLIENT_ID,
+      clientSecret: process.env.CLIENT_SECRET,
+      refreshToken: process.env.REFRESH_TOKEN,
+      oidcIssuer: OIDC_ISSUER,
+    });
+    const response = await session.fetch(
+      "https://ldp.demo-ess.inrupt.com/105177326598249077653/"
+    );
+    expect(response.status).toEqual(200);
+    await expect(response.text()).resolves.toContain("ldp:BasicContainer");
+  });
+});
+
+describe("Unauthenticated fetch", () => {
+  it("can fetch a public resource when not logged in", async () => {
+    const session = new Session();
+    const response = await session.fetch(
+      "https://ldp.demo-ess.inrupt.com/105177326598249077653/profile/card#me"
+    );
+    expect(response.status).toEqual(200);
+    await expect(response.text()).resolves.toContain(
+      "foaf:PersonalProfileDocument"
+    );
+  });
+
+  it("cannot fetch a private resource when not logged in", async () => {
+    const session = new Session();
+    const response = await session.fetch(
+      "https://ldp.demo-ess.inrupt.com/105177326598249077653/"
+    );
+    expect(response.status).toEqual(401);
+  });
+});
+
+describe("Post-logout fetch", () => {
+  it("can fetch a public resource after logging out", async () => {
+    const session = new Session();
+    await session.login({
+      clientId: process.env.CLIENT_ID,
+      clientSecret: process.env.CLIENT_SECRET,
+      refreshToken: process.env.REFRESH_TOKEN,
+      oidcIssuer: OIDC_ISSUER,
+    });
+    await session.logout();
+    const response = await session.fetch(
+      "https://ldp.demo-ess.inrupt.com/105177326598249077653/profile/card#me"
+    );
+    expect(response.status).toEqual(200);
+    await expect(response.text()).resolves.toContain(
+      "foaf:PersonalProfileDocument"
+    );
+  });
+
+  it("cannot fetch a private resource after logging out", async () => {
```

Comment:
Kinda replicates bits of other tests, but I'd have run a full sequence of steps for this particular test, i.e.:
- fetch private resource - expect 401
- login
- fetched private resource - expect 200
- logout
- fetched private resource - expect 401

Reply:
resolved in 9aa1fd7c8349f5d22f61481117f9a93d675471eb